### PR TITLE
Update list all medication

### PR DESCRIPTION
**Pull request by e0958902**
# Checklist for feature implementation and modification
Closes NIL
- [x] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [ ] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [ ] JUnit Test(s)
- Documentation
  - [x] JavaDoc for classes and methods
  - Developer Guide
    - [ ] Design (Can add as a todo statement for later on)
    - [ ] Implementation (Can add as a todo statement for later on)
    - [x] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [ ] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [ ] Code/Feature/Enhancement contribution
    - [x] Developer Guide contribution
    - [ ] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Checklist for non-implementation related PR
Closes NIL
- [x] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [x] Developer Guide contributions (if any)
  - [ ] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [ ] Miscellaneous and team-based admin tasks (if any)

# Additional Information

### CODE REVIEW

File: src/main/java/meditracker/medication/Medication.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ public void setRepeat(String repeat) {
 
     @Override
     public String toString() {
-        return getName() + " | " + getQuantity() + " | " + getDosage();
+        return getName() + " | " + getQuantity() + " | " + getDosage() + " | " +
+                getExpiryDate() + " | " + getIntakeFreq() + " | " + getRemarks();
```
Comment (T0nyLin) [inline src/main/java/meditracker/medication/Medication.java:133]: Please do not include getDosage and getIntakeFreq. We will be removing them soon.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -74,19 +84,23 @@ Overview of the `meditracker.argument` core classes:
...
+|v1.0| user              | know what medications i have to take throughout the day      | mark them as takenor not, thus not miss or forget to take what I have to take for the day                                                              |
+|v1.0| user              | edit the details in my medication list                       | ensure that its medication information is up-to-date with my medifications                                                                             |
+|v1.0| multipurpose user | Store data locally                                           | Use the application and see the data even when offline between sessions                                                                                |
+|v2.0| user              | find a to-do item by name                                    | locate a to-do without having to go through the entire list                                                                                            |
```
Comment (T0nyLin) [inline docs/DeveloperGuide.md:103]: The user stories look good! Thanks for adding them in!

**Review by T0nyLin — CHANGES_REQUESTED** [review]
All looks good! Only a minor suggestion

File: src/main/java/meditracker/medication/Medication.java (hunk 1 above)
Comment (e0958902) [reply src/main/java/meditracker/medication/Medication.java:133]: Copy that. Working on it right away.

Comment (e0958902) [reply src/main/java/meditracker/medication/Medication.java]: Resolved in [543ee25](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/88/commits/543ee25c3506852d62de3ee5073138f83ccea26c). However needed more commits such as [4deb0e6](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/88/commits/4deb0e6d086ac0b942b7b6c9cceaac0289fd8a4f) and [ad83c8c](https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/88/commits/ad83c8c2b245a9079aaee70be9f5e6992b110ac3) due to the changes done.

**Review by T0nyLin — APPROVED** [review]
LGTM

**Pull request merged**